app/views/handlers/list_collector.py

```diff
@@ -55,5 +69,22 @@ def handle_post(self):
             # wtforms Form parents are not discoverable in the 2.3.3 implementation
             self.questionnaire_store_updater.update_answers(self.form.data)
             self.questionnaire_store_updater.save()
-        else:
+        elif not self.rendered_block["repeating_blocks"] or (self.rendered_block["repeating_blocks"] and self._is_list_collector_complete()):
             return super().handle_post()
+
+    def _is_list_collector_complete(self):
+        list_model = self._questionnaire_store.list_store.get(self.rendered_block["for_list"])
+        return all(
+            self.questionnaire_store_updater.is_section_complete(section_id=self.current_location.section_id, list_item_id=list_item_id)
+            for list_item_id in list_model.items)
+
+    def _get_incomplete_repeating_block_ids(self) -> Generator:
+        list_model = self._questionnaire_store.list_store.get(self.rendered_block["for_list"])
+        repeating_blocks = self.rendered_block.get("repeating_blocks")
```

Comment:
Do this first and return if we don't have a repeating block?. Doing so makes this method reusable in a lot of places.

Reply:
moved it, good call 👍